ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala

```diff
@@ -73,39 +80,70 @@ private[kvutils] object InputsAndEffects {
               addLogEntryInput(logEntryInputs, fetch.coid),
               addStateInput(stateInputs, fetch.coid)
             )
-          case create: NodeCreate[_, _] =>
-            (logEntryInputs, stateInputs)
+          case create: NodeCreate[ContractId, VersionedValue[ContractId]] =>
+            (
+              logEntryInputs,
+              create.key.fold(stateInputs) { keyWithM =>
+                contractKeyToStateKey(GlobalKey(
+                  create.coinst.template,
+                  forceAbsoluteContractIds(keyWithM.key))) :: stateInputs
+              }
+            )
           case exe: NodeExercises[_, ContractId, _] =>
             (
               addLogEntryInput(logEntryInputs, exe.targetCoid),
               addStateInput(stateInputs, exe.targetCoid)
```

Comment:
If the exercise consumes the contract and that contract has a key, then you should also include this in the key update effects as the key will be unassigned afterwards.